pkg/k8shandler/forwarding.go

```diff
@@ -48,7 +49,26 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 
 	clusterRequest.ForwardingSpec = clusterRequest.normalizeLogForwarding(clusterRequest.cluster.Namespace, clusterRequest.cluster)
 	generator, err := forwarding.NewConfigGenerator(clusterRequest.cluster.Spec.Collection.Logs.Type, clusterRequest.includeLegacyForwardConfig(), clusterRequest.includeLegacySyslogConfig())
-	return generator.Generate(&clusterRequest.ForwardingSpec)
+	generatedConfig, err := generator.Generate(&clusterRequest.ForwardingSpec)
+
+	if err != nil {
+		return "",
+			clusterRequest.UpdateCondition(
```

Comment:
as we talked about on slack, we should not have the return here since we will obscure the original `err`.
please update the if/else to just call `clusterRequest.UpdateCondition` as it is currently structured, and leave the return for the end of the function as `return generatedConfig, err`

Reply:
Late evening discussion missed to get persistent in my brain

Reply:
> i thought we agreed on slack, that we should not have the return here since we will obscure the original `err`.

Regarding returning the error back to the reconciliation loop, I believe the reason no to return was a good one. We basically log out the error as a warning and add the DeadEnd Condition to the CR. We stop/pause the reconciliation loop because manual user intervention is in need here.

FYI, i put back the old behavior for this backport, because otherwise we are keeping reconciling the DeadEnd Condition endlessly. WDYT?

In addition, I put the missing error handling for creating the generator.

Reply:
the concern i have with us returning the result of the `clusterRequest.UpdateCondition` error instead of `err` is that we are hiding an error that who ever is calling this may want to handle in a specific way. I think we should still return the error, but also adjust the error handling of whomever calls the `generateCollectorConfig()` function.

Reply:
I believe the following things are orthogonal:
* Returning the original error because generateConfig did not yield a valid configuration
* Reconcile fluentd with an empty configuration w/o crashlooping and in addition having a DeadEnd condition that indicates manual user intervention on the CR.

Returning the original error will keep the reconciliation trying to generate a config while fluentd remains on crashlooping w/o a config.

Reply:
what about doing this:
1. return the original error, because that is a better code practice than letting the caller think that everything was alright
2. adjust the caller to not keep retrying in the case that it was unable to generate a config (if we don't have a config, then don't spin up a fluentd daemonset, and have a condition explain this)

Reply:
@ewolinetz Exactly the opposite is the case of the linked BZ. According to @sichvoge we support a fluentd-only setup in clusterlogging and that used to work before LF introduction. However until now the destiny of spining up a fluentd daemonset and generating a configuration is tightly bound. Therefore if later does not succeed - as it is the case of the BZ not providing any LF or logstore bits - I believe we need to shallow the original error into a log entry and in addition singal to the user the status condition "DeadEnd".